### ISSUE TITLE
Error building extension 'nvdiffrast_plugin' ninja: build stopped: subcommand failed.

### ISSUE THREAD

**Issue by amoeet** [flowtyone/ComfyUI-Flowty-CRM#14]
![`X9HRSA$U2XR6DN3OEBCT7V](https://github.com/flowtyone/ComfyUI-Flowty-CRM/assets/69706187/5fc7824d-8b6e-4ee2-aad8-c1ee4b048b19)

![QQ图片20240317180309](https://github.com/flowtyone/ComfyUI-Flowty-CRM/assets/69706187/9cfa359f-9c19-4e48-b3e0-4def565a60d2)

Error occurred when executing CRMModelerCuda:

Error building extension 'nvdiffrast_plugin': [1/16] cl /showIncludes -DTORCH_EXTENSION_NAME=nvdiffrast_plugin -DTORCH_API_INCLUDE_EXTENSION_H -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\torch\csrc\api\include -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\TH -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\THC "-IC:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.1\include" -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\Include -D_GLIBCXX_USE_CXX11_ABI=0 /MD /wd4819 /wd4251 /wd4244 /wd4267 /wd4275 /wd4018 /wd4190 /wd4624 /wd4067 /wd4068 /EHsc /std:c++17 -DNVDR_TORCH -c H:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\nvdiffrast\common\common.cpp /Focommon.o
用于 x64 的 Microsoft (R) C/C++ 优化编译器 19.39.33519 版
版权所有(C) Microsoft Corporation。保留所有权利。

注意: 包含文件: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.1\include\cuda_runtime.h
注意: 包含文件: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.1\include\crt/host_config.h
...
Omit 30000 rows
...
注意: 包含文件: H:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\torch\csrc\api\include\torch/python.h
注意: 包含文件: H:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\torch/csrc/Device.h
注意: 包含文件: H:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\torch/csrc/python_headers.h
H:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\torch/csrc/python_headers.h(12): fatal error C1083: 无法打开包括文件: “python.h”: No such file or directory
[15/16] C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.1\bin\nvcc --generate-dependencies-with-compile --dependency-output texture.cuda.o.d -Xcudafe --diag_suppress=dll_interface_conflict_dllexport_assumed -Xcudafe --diag_suppress=dll_interface_conflict_none_assumed -Xcudafe --diag_suppress=field_without_dll_interface -Xcudafe --diag_suppress=base_class_has_different_dll_interface -Xcompiler /EHsc -Xcompiler /wd4068 -Xcompiler /wd4067 -Xcompiler /wd4624 -Xcompiler /wd4190 -Xcompiler /wd4018 -Xcompiler /wd4275 -Xcompiler /wd4267 -Xcompiler /wd4244 -Xcompiler /wd4251 -Xcompiler /wd4819 -Xcompiler /MD -DTORCH_EXTENSION_NAME=nvdiffrast_plugin -DTORCH_API_INCLUDE_EXTENSION_H -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\torch\csrc\api\include -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\TH -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\torch\include\THC "-IC:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.1\include" -IH:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\Include -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_86,code=compute_86 -gencode=arch=compute_86,code=sm_86 -std=c++17 -DNVDR_TORCH -lineinfo -c H:\AI_Draw\ComfyUIxBlender\Blender_ComfyUI\python_embeded\lib\site-packages\nvdiffrast\common\texture.cu -o texture.cuda.o
texture.cu

tmpxft_00002a70_00000000-10_texture.cudafe1.cpp

ninja: build stopped: subcommand failed.

Platform: Windows
Python version: 3.10.9
Cuda compilation tools, release 12.1, V12.1.66
Build cuda_12.1.r12.1/compiler.32415258_0

Package                        Version
------------------------------ ------------------
absl-py                        1.4.0
accelerate                     0.28.0
addict                         2.4.0
aiofiles                       23.1.0
aiohttp                        3.8.4
aiosignal                      1.3.1
albumentations                 1.3.1
altair                         5.0.0
annotated-types                0.6.0
antlr4-python3-runtime         4.9.3
anyio                          3.6.2
arabic-reshaper                3.0.0
argostranslate                 1.9.1
asttokens                      2.4.1
astunparse                     1.6.3
async-timeout                  4.0.2
asyncer                        0.0.2
attrs                          22.2.0
audioread                      3.0.1
auto_gptq                      0.7.1
Babel                          2.14.0
basicsr                        1.4.2
beautifulsoup4                 4.12.2
blend-modes                    2.1.0
blinker                        1.7.0
boltons                        23.0.0
bpy                            3.6.0
cachetools                     5.3.1
certifi                        2022.12.7
cffi                           1.15.1
chardet                        3.0.4
charset-normalizer             3.1.0
click                          8.1.3
clip-interrogator              0.6.0
cmake                          3.27.2
color-matcher                  0.5.0
colorama                       0.4.6
coloredlogs                    15.0.1
colorlog                       6.8.0
colour-science                 0.4.4
compel                         2.0.0
contourpy                      1.0.7
cryptography                   41.0.7
cssselect2                     0.7.0
cstr                           0.1.0
ctranslate2                    3.20.0
cupy-cuda11x                   12.3.0
cycler                         0.11.0
Cython                         3.0.0
dataclasses-json               0.6.4
datasets                       2.18.0
ddt                            1.7.0
decorator                      4.4.2
deep-translator                1.11.4
deepdiff                       6.7.1
Deprecated                     1.2.14
diffusers                      0.24.0
dill                           0.3.8
diskcache                      5.6.3
distro                         1.8.0
docstring-parser               0.15
docutils                       0.20.1
easydict                       1.10
einops                         0.7.0
embreex                        2.17.7.post4
evalidate                      2.0.2
exceptiongroup                 1.2.0
executing                      2.0.1
face-recognition-models        0.3.0
facexlib                       0.3.0
fairscale                      0.4.13
fastapi                        0.94.0
fastrlock                      0.8.2
ffmpeg                         1.4
ffmpeg-python                  0.2.0
ffmpy                          0.3.0
filelock                       3.10.0
filetype                       1.2.0
filterpy                       1.4.5
fire                           0.5.0
Flask                          3.0.0
flatbuffers                    24.3.7
fonttools                      4.39.2
frozenlist                     1.3.3
fsspec                         2023.12.2
ftfy                           6.1.1
future                         0.18.3
fvcore                         0.1.5.post20221221
gast                           0.4.0
gdown                          4.7.1
gekko                          1.0.7
ghp-import                     2.1.0
gitdb                          4.0.10
GitPython                      3.1.31
glitch-this                    1.0.2
google-ai-generativelanguage   0.4.0
google-api-core                2.17.1
google-auth                    2.22.0
google-auth-oauthlib           0.4.6
google-generativeai            0.4.0
google-pasta                   0.2.0
googleapis-common-protos       1.62.0
googletrans                    3.1.0a0
googletrans-py                 4.0.0
gradio                         3.32.0
gradio_client                  0.2.4
griffe                         0.42.0
grpcio                         1.62.0
grpcio-status                  1.62.0
h11                            0.14.0
h2                             4.1.0
h5py                           3.9.0
hpack                          4.0.0
hstspreload                    2023.1.1
httpcore                       1.0.2
httpx                          0.25.2
huggingface-hub                0.20.3
humanfriendly                  10.0
hyperframe                     6.0.1
icecream                       2.1.3
idna                           2.10
ImageHash                      4.3.1
imageio                        2.27.0
imageio-ffmpeg                 0.4.9
img2texture                    1.0.6
importlib-metadata             6.8.0
insightface                    0.7.3
iopath                         0.1.10
itsdangerous                   2.1.2
jax                            0.4.23
Jinja2                         3.1.2
joblib                         1.2.0
jsonschema                     4.17.3
keras                          2.12.0
Keras-Preprocessing            1.1.2
keyframed                      0.3.15
kiui                           0.2.5
kiwisolver                     1.4.4
kornia                         0.7.2
kornia_rs                      0.1.1
lark                           1.1.7
lazy_loader                    0.2
libclang                       16.0.6
librosa                        0.10.1
lightning-utilities            0.8.0
linkify-it-py                  2.0.2
llama-cpp-agent                0.0.17
llama_cpp_python               0.2.44
llvmlite                       0.40.0
lmdb                           1.4.1
loguru                         0.7.2
lpips                          0.1.4
lxml                           4.9.2
mapbox-earcut                  1.0.1
Markdown                       3.5.2
markdown-it-py                 2.2.0
MarkupSafe                     2.1.2
marshmallow                    3.21.1
matplotlib                     3.8.0
matrix-client                  0.4.0
mdit-py-plugins                0.3.3
mdurl                          0.1.2
mediapipe                      0.9.1.0
mergedeep                      1.3.4
mkdocs                         1.5.3
mkdocs-autorefs                1.0.1
mkdocs-material                9.5.13
mkdocs-material-extensions     1.3.1
mkdocstrings                   0.24.1
mkdocstrings-python            1.9.0
ml-dtypes                      0.3.2
mmcv                           2.0.0
mmdet                          3.0.0
mmengine                       0.7.3
model-index                    0.1.11
moviepy                        1.0.3
mpmath                         1.3.0
msgpack                        1.0.7
msvc-runtime                   14.34.31931
multidict                      6.0.4
multiprocess                   0.70.16
mypy-extensions                1.0.0
networkx                       3.0
ninja                          1.11.1.1
numba                          0.57.0
numexpr                        2.8.7
numpy                          1.24.4
nvdiffrast                     0.3.1
oauthlib                       3.2.2
objprint                       0.2.3
omegaconf                      2.3.0
onnx                           1.15.0
onnxruntime                    1.17.1
onnxruntime-gpu                1.17.1
open-clip-torch                2.7.0
openai                         1.3.7
opencv-contrib-python          4.9.0.80
opencv-contrib-python-headless 4.9.0.80
opencv-python                  4.9.0.80
opencv-python-headless         4.9.0.80
openmim                        0.3.7
opt-einsum                     3.3.0
ordered-set                    4.1.0
orjson                         3.9.0
packaging                      24.0
paginate                       0.5.6
pandas                         2.0.2
pathspec                       0.12.1
peft                           0.9.0
piexif                         1.1.3
pilgram                        1.2.1
Pillow                         10.1.0
pip                            24.0
pixelsort                      1.0.1
platformdirs                   3.6.0
pooch                          1.7.0
portalocker                    2.7.0
prettytable                    3.6.0
proglog                        0.1.10
proto-plus                     1.23.0
protobuf                       3.20.3
psutil                         5.9.4
py-cpuinfo                     3.3.0
pyarrow                        15.0.1
pyarrow-hotfix                 0.6
pyasn1                         0.5.0
pyasn1-modules                 0.3.0
pycocoevalcap                  1.2
pycocotools                    2.0.6
pycollada                      0.7.2
pycparser                      2.21
pydantic                       2.6.4
pydantic_core                  2.16.3
pydub                          0.25.1
pygame                         2.5.0
pygit2                         1.11.1
pygltflib                      1.16.2
Pygments                       2.17.2
PyMatting                      1.1.8
pymdown-extensions             10.7.1
pymeshlab                      2023.12.post1
pynvml                         11.5.0
pyOpenSSL                      23.3.0
pyparsing                      3.0.9
pypng                          0.20220715.0
pyproject-metadata             0.7.1
pyre-extensions                0.0.23
pyreadline3                    3.4.1
pyrsistent                     0.19.3
pySmartDL                      1.3.4
PySocks                        1.7.1
python-bidi                    0.4.2
python-dateutil                2.8.2
python-multipart               0.0.6
pythonperlin                   0.0.3
pytorch-lightning              2.0.0
pytz                           2023.3
PyWavelets                     1.4.1
pywin32                        306
PyYAML                         6.0
pyyaml_env_tag                 0.1
qrcode                         7.4.2
qudida                         0.0.4
regex                          2022.10.31
rembg                          2.0.50
reportlab                      3.6.12
requests                       2.28.2
requests-oauthlib              1.3.1
requirements-parser            0.5.0
rfc3986                        1.5.0
rich                           13.4.1
rich-argparse                  1.4.0
rouge                          1.0.1
rsa                            4.9
Rtree                          1.1.0
sacremoses                     0.0.53
safetensors                    0.4.2
scikit-build                   0.17.6
scikit_build_core              0.8.1
scikit-image                   0.20.0
scikit-learn                   1.3.2
scipy                          1.11.3
seaborn                        0.12.2
segment-anything               1.0
semantic-version               2.10.0
sentencepiece                  0.1.99
setuptools                     69.2.0
shapely                        2.0.1
simple-lama-inpainting         0.1.2
simpleeval                     0.9.13
six                            1.16.0
smmap                          5.0.0
sniffio                        1.3.0
sortedcontainers               2.4.0
soundfile                      0.12.1
soupsieve                      2.4.1
soxr                           0.3.7
stanza                         1.1.1
starlette                      0.26.1
svg.path                       6.3
svglib                         1.5.1
sympy                          1.12
symusic                        0.4.2
tabulate                       0.9.0
tb-nightly                     2.12.0a20230209
tensorboard                    2.12.0
tensorboard-data-server        0.7.2
tensorboard-plugin-wit         1.8.1
tensorflow                     2.12.0
tensorflow-estimator           2.12.0
tensorflow-intel               2.12.0
tensorflow-io-gcs-filesystem   0.31.0
termcolor                      2.3.0
terminaltables                 3.1.10
threadpoolctl                  3.2.0
tifffile                       2023.3.21
timm                           0.6.7
tinycss2                       1.2.1
tokenizers                     0.15.2
tomesd                         0.1.3
tomli                          2.0.1
toolz                          0.12.0
torch                          2.1.2+cu121
torchaudio                     2.1.2+cu121
torchdiffeq                    0.2.3
torchmetrics                   0.11.4
torchsde                       0.2.6
torchvision                    0.16.2+cu121
tqdm                           4.65.0
trampoline                     0.1.2
transformers                   4.38.2
trimesh                        4.0.8
typer                          0.9.0
typer-config                   1.4.0
types-setuptools               68.1.0.0
typing                         3.7.4.3
typing_extensions              4.9.0
typing-inspect                 0.8.0
tzdata                         2023.3
uc-micro-py                    1.0.2
ultralytics                    8.0.142
urllib3                        1.26.15
uvicorn                        0.22.0
varname                        0.13.0
watchdog                       3.0.0
wcwidth                        0.2.6
webencodings                   0.5.1
websocket-client               1.7.0
websockets                     11.0.3
Werkzeug                       3.0.1
wget                           3.2
wheel                          0.40.0
win32-setctime                 1.1.0
wrapt                          1.14.1
xatlas                         0.0.9
xformers                       0.0.23.post1
xxhash                         3.4.1
yacs                           0.1.8
yapf                           0.32.0
yarl                           1.8.2
zipp                           3.16.2
zstandard                      0.21.0

 
I installed nvdifrest using the following issuecomment

https://github.com/flowtyone/ComfyUI-Flowty-CRM/issues/4#issuecomment-2002377703

Added ninja.exe to the path of the system environment variable

Visual Studio Installer installed the C++generation tool for VS 2022

Please help me take a look. What else do I need to do to fix this error? I feel like I'm just one step away

**Comment by 0ZXJ1120** [comment]
same error
'Error building extension 'nvdiffrast_plugin'

**Comment by Publicoprivado** [comment]
I have the same error driving me nuts, keep getting the same error.

**Comment by gerroon** [comment]
I have the same issue :( damn, I really hope someone knows how to solve this

**Comment by balintligeti** [comment]
same issue

**Comment by CharliGuo** [comment]
@amoeet 楼主解决了吗？求一个solution
